lib/cn/spill.c

```diff
@@ -251,8 +251,6 @@ kv_spill(struct cn_compaction_work *w)
      * readahead buffer).
      */
     direct_read_len = w->cw_rp->cn_compact_vblk_ra;
-    if (direct_read_len < 32 * 1024)
-        direct_read_len = 32 * 1024;
```

Comment:
Why did you eliminate this?

Reply:
Because I changed the bounds on cn_compact_vblk_ra. If the value can never be less than MPOOL_MBLOCK_SIZE_DEFAULT, then why not use it as ps_min.

Reply:
I'll need to study this to see if it makes sense that direct-read-len is now dependent upon cn_compact_vblk_ra.  Dunno of the top of my head...

Reply:
Note that the read-ahead values are now the min(mblocksz of all configured media classes) at this point within HSE.

Reply:
Ok, this is coming back to me.  The vblock readahead used to be 256K bytes, which is small because a spill could be reading values from any of several hundred vblocks during any given spill.  The readahead code initiates a readahead as the values in any given buffer are consumed such that half the buffer has been drained (IIRC), so as to try and keep slightly ahead of the reader yet minizmize the memory foot print to perform a spill.  This particular code was trying to deal with values that are larger than the readhead buffer (such large values a typically rare).
So what mblocksz now, typically?

Reply:
HSE currently only supports an mblock size of 32 MB (MPOOL_MBLOCK_SIZE_DEFAULT). There is a check in ikvdb.c.

Reply:
MPOOL_MBLOCK_SIZE_DEFAULT is 32*1024*1024.  This is 32*1024.  Need to fix `.ps_min` in the param spec.

Reply:
Whoops...32 K

Reply:
Then this seems really strange because these read ahead parameters are based on the min(mblocksz of all configured media classes), so it will be in the MB of values regardless because HSE only supports 32MB mblock size. Confused to say the least.

Reply:
The readahead values shouldn't be at all linked to mblock sizes...  Where do you see that?  Per my .ps_max comment, the vblock readahead buffer should never be larger than 2M (but really 2x4M because there are actually two buffers within one allocation, one being looked at by spill and the other undergoing read, where they swap when the spill-read buffer is exhausted).  But .ps_max should set the max to 2M and this code will double it.